src/compiler/utilities.ts

```diff
@@ -6288,6 +6290,21 @@ export function getPossibleOriginalInputExtensionForExtension(path: string) {
         [Extension.Tsx, Extension.Ts, Extension.Jsx, Extension.Js];
 }
 
+/** @internal */
+export function getPossibleOriginalInputPathWithoutChangingExt(

```

Comment:
this is kinda a reverse of `getOutputPathWithoutChangingExt`: https://github.com/microsoft/TypeScript/blob/3fca8c842a42bb43ba5a86dcce1b7fb0eae6c7a4/src/compiler/emitter.ts#L542-L554